Spire controller manager upgrade (#8)
spiffe/helm-charts-hardened: charts/spire/README.md

| file://./charts/spire-server | spire-server | 0.1.0 |
| file://./charts/tornjak-frontend | tornjak-frontend | 0.1.0 |

## Install notes

To do a quick non production install:

> kubectl create namespace spire-server
> helm install -n spire-server spire-crds charts/spire-crds
> helm install -n spire-server spire charts/spire

Reviewer: ```suggestion
```shell
kubectl create namespace spire-system
helm install -n spire-system spire-crds charts/spire-crds
helm install -n spire-system spire charts/spire
```
```

> For production installs, please see examples/production/

Reviewer: ```suggestion
For production installs, please see [the production example](examples/production/).
```

## Upgrade notes

0.14.X:

If coming from a chart version before 0.14.0, you must relabel your crds to switch to using the new spire-crds chart. To migrate to the spire-crds chart
run the following:

> ```

Reviewer: ```suggestion
```shell
```

> kubectl label crd "clusterfederatedtrustdomains.spire.spiffe.io" "app.kubernetes.io/managed-by=Helm"
> kubectl annotate crd "clusterfederatedtrustdomains.spire.spiffe.io" "meta.helm.sh/release-name=spire-crds"
> kubectl annotate crd "clusterfederatedtrustdomains.spire.spiffe.io" "meta.helm.sh/release-namespace=spire-server"
> kubectl label crd "clusterspiffeids.spire.spiffe.io" "app.kubernetes.io/managed-by=Helm"
> kubectl annotate crd "clusterspiffeids.spire.spiffe.io" "meta.helm.sh/release-name=spire-crds"
> kubectl annotate crd "clusterspiffeids.spire.spiffe.io" "meta.helm.sh/release-namespace=spire-server"
> kubectl label crd "controllermanagerconfigs.spire.spiffe.io" "app.kubernetes.io/managed-by=Helm"
> kubectl annotate crd "controllermanagerconfigs.spire.spiffe.io" "meta.helm.sh/release-name=spire-crds"
> kubectl annotate crd "controllermanagerconfigs.spire.spiffe.io" "meta.helm.sh/release-namespace=spire-server"
> 
> #Update -n with the namespace your installing the spire chart into.
> helm install -n spire-server spire-crds charts/spire-crds

Reviewer: ```suggestion
# Set the namespace to the namespace you are installing the spire chart.
namespace=spire-system
kubectl label crd "clusterfederatedtrustdomains.spire.spiffe.io" "app.kubernetes.io/managed-by=Helm"
kubectl annotate crd "clusterfederatedtrustdomains.spire.spiffe.io" "meta.helm.sh/release-name=spire-crds"
kubectl annotate crd "clusterfederatedtrustdomains.spire.spiffe.io" "meta.helm.sh/release-namespace=${namespace}"
kubectl label crd "clusterspiffeids.spire.spiffe.io" "app.kubernetes.io/managed-by=Helm"
kubectl annotate crd "clusterspiffeids.spire.spiffe.io" "meta.helm.sh/release-name=spire-crds"
kubectl annotate crd "clusterspiffeids.spire.spiffe.io" "meta.helm.sh/release-namespace=spire-server"
kubectl label crd "controllermanagerconfigs.spire.spiffe.io" "app.kubernetes.io/managed-by=Helm"
kubectl annotate crd "controllermanagerconfigs.spire.spiffe.io" "meta.helm.sh/release-name=spire-crds"
kubectl annotate crd "controllermanagerconfigs.spire.spiffe.io" "meta.helm.sh/release-namespace=${namespace}"

helm install -n "${namespace}" spire-crds charts/spire-crds
```

Introduced a variable for the namespace as it is also used in the annotations, which also has to be aligned.

Author: This one is a bit trickier to deal with...

I spelled it all out,so the commands would work no matter which os you use. With the variables, then it only works with linux/macos/windows but only with bash. If the user is using cmd/powershell, it breaks.

Should we have them edit the strings in place and then only need one set of instructions, or have two, and include a powershell example?

Reviewer: OK. In that case maybe just move the comment to the top to cover the whole block.

```suggestion
# replace the spire-server namespace in below commands with the namespace you have been using to install spire-server.

kubectl label crd "clusterfederatedtrustdomains.spire.spiffe.io" "app.kubernetes.io/managed-by=Helm"
kubectl annotate crd "clusterfederatedtrustdomains.spire.spiffe.io" "meta.helm.sh/release-name=spire-crds"
kubectl annotate crd "clusterfederatedtrustdomains.spire.spiffe.io" "meta.helm.sh/release-namespace=spire-server"
kubectl label crd "clusterspiffeids.spire.spiffe.io" "app.kubernetes.io/managed-by=Helm"
kubectl annotate crd "clusterspiffeids.spire.spiffe.io" "meta.helm.sh/release-name=spire-crds"
kubectl annotate crd "clusterspiffeids.spire.spiffe.io" "meta.helm.sh/release-namespace=spire-server"
kubectl label crd "controllermanagerconfigs.spire.spiffe.io" "app.kubernetes.io/managed-by=Helm"
kubectl annotate crd "controllermanagerconfigs.spire.spiffe.io" "meta.helm.sh/release-name=spire-crds"
kubectl annotate crd "controllermanagerconfigs.spire.spiffe.io" "meta.helm.sh/release-namespace=spire-server"
helm install -n spire-server spire-crds charts/spire-crds
```

| file://./charts/spire-server | spire-server | 0.1.0 |
| file://./charts/tornjak-frontend | tornjak-frontend | 0.1.0 |

## Install notes

To do a quick non production install:

```shell
kubectl create namespace spire-system
helm install -n spire-system spire-crds charts/spire-crds
helm install -n spire-system spire charts/spire

For production installs, please see [the production example](examples/production/).

## Upgrade notes

0.14.X:

If coming from a chart version before 0.14.0, you must relabel your crds to switch to using the new spire-crds chart. To migrate to the spire-crds chart
run the following:

```shell

> # replace the spire-server namespace in below commands with the namespace you have been using to install spire-server.

Author: ```suggestion
# Replace the spire-server namespace in the commands below with the namespace you want to install the spire-crds chart in.
```
